sdk/eventhub/event-hubs/test/receiver.spec.ts

```diff
@@ -72,6 +72,96 @@ describe("EventHub Receiver", function(): void {
     });
   });
 
+  describe("receiver names", function(): void {
+    it("are generated and unique if not provided", async function(): Promise<void> {
+      const partitionIds = await client.getPartitionIds();
+      partitionIds.length.should.be.greaterThan(1, "Event Hub should have more than 1 partition to run this test.");
+
+      const receiveHandlers: ReceiveHandler[] = [];
+      const receiverNames: Set<string> = new Set();
+
+      for (const partitionId of partitionIds) {
+        const handler = client.receive(
+          partitionId,
+          () => {},
+          () => {},
+          {
+            eventPosition: EventPosition.fromEnd()
+          }
+        );
+
+        should.equal(typeof handler.name === "string", true, "Receiver name is not a string.");
+
+        receiverNames.add(handler.name);
+        receiveHandlers.push(handler);
+      }
+
+      should.equal(receiverNames.size, partitionIds.length, "Unexpected number of receiver names found.");
+
+      // wait for all receivers to be open before stopping them.
```

Comment:
why wait? why not add an afterEach that closes the client?

Reply:
I found a possible bug in our library (will log an issue). If I call stop on the receiveHandler before the cbs link has been established, then client.close will throw an error because it attempts to call close on the link that doesn't yet exist.

Reply:
#6512 logged